tests/testthat/test-modspec.R

```diff
@@ -587,3 +587,64 @@ test_that("INPUT block", {
   expect_equal(tagdf$name, c("CL", "V2"))
   expect_equal(tagdf$tag, rep("input", 2))
 })
+
+test_that("Reserve names in cpp dot gh-1159", {
+  temp <- '$param {param}\n$main\n{main};\n$table\n{table}'
```

Comment:
Okay as is, but I think it'd be worth testing `init`, `omega_labels`, and `sigma_labels` too given that `check_cpp_dot` repeats some logic for those.

Reply:
I refactored these tests to look at parameters, compartments, omega labels and sigma labels. The "problem" code is placed in preamble, main, ode, and table.